File: server/lib/validation.dart
/// Small input-validation helpers shared by the auth routes. Mirrors
/// `server/config/validation.js`; keep the two in lockstep.

/// Minimum accepted password length (characters).
const int minPasswordLength = 8;

final RegExp _emailRegExp = RegExp(r'^[^\s@]+@[^\s@]+\.[^\s@]+$');

/// Pragmatic email-format check (not a full RFC 5322 validator): a single `@`
/// separating a non-empty local part and a domain with a dot, no whitespace.
bool isValidEmail(String? email) =>
    email != null && _emailRegExp.hasMatch(email.trim());

/// A password is accepted when it is at least [minPasswordLength] chars.
bool isStrongPassword(String? password) =>
    password != null && password.length >= minPasswordLength;

File: server/config/validation.js
/**
 * Small input-validation helpers shared by the auth routes.
 */

/** Minimum accepted password length (characters). */
export const MIN_PASSWORD_LENGTH = 8;

/**
 * Pragmatic email-format check: a single `@` separating a non-empty local part
 * and a domain with a dot, and no whitespace. Not a full RFC 5322 validator —
 * just enough to reject obviously malformed input.
 */
export function isValidEmail(email) {
  if (typeof email !== 'string') return false;
  return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email.trim());
}

/** A password is accepted when it is at least [MIN_PASSWORD_LENGTH] chars. */
export function isStrongPassword(password) {
  return typeof password === 'string' && password.length >= MIN_PASSWORD_LENGTH;
}
